### DISCUSSION TITLE
Week4 作業問題 關於API文件的參數

### DISCUSSION THREAD

**Discussion by neniz77** [Lidemy/forum#115] [程式導師實驗計畫] [answered]
<!DOCTYPE html>

關於 hw1和2 的 API 說明，我不太明白為什麼 _limit 和 name 兩者為什麼都放在「參數」這欄。

在我的理解裡面，limit 和 name 的使用方法不太ㄧ樣，limit 會放在 path 後面，name 則是放在 request 的 body 裡面。所以想請問這兩者的共通點在哪？為什麼可以一起放在參數這欄裡面呢？或者說是想請問為什麼這樣寫，使用者會知道 name 這個參數應該是放在 body 裡面？

說明 | Method | path | 參數 | 範例
-- | -- | -- | -- | --
獲取所有書籍 | GET | /books | _limit:限制回傳資料數量 | /books?_limit=5
獲取單一書籍 | GET | /books/:id | 無 | /books/10
新增書籍 | POST | /books | name: 書名 | 無
刪除書籍 | DELETE | /books/:id | 無 | 無
更改書籍資訊 | PATCH | /books/:id | name: 書名 | 無

**Comment by allenliaooooooo** [comment, marked as answer]
我覺得老師只是為了簡化表格（或是懶？😂）
主要是想表達這個動作你會需要操作到的那個參數的名稱叫什麼
我的理解這兩個東西是不一樣的東西沒錯

**Reply by yichennnn36** [reply]
老師在額外補充影片中好像也有提到說，他的表格寫的比較簡略，例如說好像也沒有寫到要用什麼格式帶資料到 body？
不知道有沒有記錯

**Reply by allenliaooooooo** [reply]
嗯嗯，所以這部分要自己去查資料研究

**Reply by aszx87410** [reply]
對，是為了簡化表格沒錯，其實應該要寫得更詳細的，不然其實要靠自己試出來

**Reply by neniz77** [reply]
原來如此，謝謝大家的回覆～
